### DISCUSSION TITLE
Removing rustfmt from CI checks

### DISCUSSION THREAD

**Discussion by maximeborges** [stm32-rs/stm32f7xx-hal#183] [General]
`rustfmt` is a great tool to format code, but lakes some flexibility when using chained function calls.
When configuring peripheral registers, it looks like it would make more sense to have something pairs of `bit().value()` on each line like the following:
```rust
w.te().enabled()
 .re().enabled()
 .ue().enabled()
 .m1().m0()
 .m0().bit(!matches!(config.parity, Parity::ParityNone))
 .pce().bit(!matches!(config.parity, Parity::ParityNone))
 .ps().bit(matches!(config.parity, Parity::ParityOdd))
```
more than what `rustfmt` would format:
```rust
w.te()
    .enabled()
    .re()
    .enabled()
    .ue()
    .enabled()
    .m1()
    .m0()
    .m0()
    .bit(!matches!(config.parity, Parity::ParityNone))
    .pce()
    .bit(!matches!(config.parity, Parity::ParityNone))
    .ps()
    .bit(matches!(config.parity, Parity::ParityOdd))
```

We could also argue about using the first `bit().value()` pair on the first line and align all the following ones with only one space, or moving the first one to the next line and using 4 spaces as everywhere else, i.e.:
```rust
w
    .te().enabled()
    .re().enabled()
    .ue().enabled()
    .m1().m0()
    .m0().bit(!matches!(config.parity, Parity::ParityNone))
    .pce().bit(!matches!(config.parity, Parity::ParityNone))
    .ps().bit(matches!(config.parity, Parity::ParityOdd))
```

Something that could be done would be to only run `rustfmt` on the CI for the lines that changed, but not throwing errors and letting the reviewer decides if the formatting would make sense to be applied or not. Something in the spirit of [rustfmt-format-diff](https://github.com/emilio/rustfmt-format-diff) could do the job.

**Comment by eldruin** [comment]
I agree with you and have removed rustfmt from the CI in #184

**Reply by maximeborges** [reply]
What's your opinion about still having `rustfmt` running on diff just to catch obvious bad formatting in CI without enforcing it, so the reviewer can still check it before merging?

**Reply by maximeborges** [reply]
pinging @mvertescher

**Reply by eldruin** [reply]
Yes, if that works it would be a good idea.
